jetstream/analysis.py

```diff
@@ -315,15 +347,22 @@ def validate(self) -> None:
         for v in self.config.metrics.values():
             metrics |= {m.metric for m in v}
 
-        sql = exp.build_query_template(
-            metrics,
+        enrollments_sql = exp.build_enrollments_query(
             limits,
             "normandy",
             self.config.experiment.enrollment_query,
             self.config.experiment.segments,
-        ).format(results_table="results_table")
+        )
+
+        dry_run_query(enrollments_sql)
+
+        metrics_sql = exp.build_metrics_query(
+            metrics,
+            limits,
+            "enrollments_table",  # todo: fails because table doesn't exist
```

Comment:
Haven't had a chance to read through the second commit yet but maybe it's worth splitting out the first part; I think we need a solution to this before we can upgrade mozanalysis.